apps/frontend/src/libs/hooks/use-pagination/use-pagination.hook.ts

```diff
@@ -61,14 +60,17 @@ const usePagination: UsePagination = ({
 	const onPageChange = useCallback(
 		(newPage: number) => {
 			const totalPages = calculateTotalPages(pageSize, totalItemsCount);
+			const validPage = Math.min(newPage, totalPages);
```

Comment:
we also need to make sure it's more than initial page
```suggestion
			const validPage = Math.max(FIRST_PAGE, Math.min(newPage, totalPages));
```

Reply:
Oh sure, my bad!
I just saw this check in `calculateTotalPages` and didn't notice it's not related.